active/AnimatedIcon/AnimatedIconSpec.md

```diff
@@ -0,0 +1,282 @@
+
+Also, AnimatedVisualPlayer does not inherit from IconElement so it cannot be used in places that
+need one, like 
+[NavigationViewItem.Icon](https://docs.microsoft.com/uwp/api/Windows.UI.Xaml.Controls.NavigationViewItem.Icon)
+or in a button.
+
+This spec introduces an AnimatedIcon IconElement which reduces, or ideally eliminates, the amount of code-behind needed to
+implement most scenarios, and it can be used in WinUI controls that support an IconElement.
+
+# How to design animations for AnimatedIcon
+
+## Creating animated content with Lottie 
+
+1. Download the Lottie file for the icon you are looking to add to your application.
+2. Run the Lottie file through LottieGen. The LottieGen will output the file that you will add to your project that will include the markers needed for animated icon to map the correct state transitions to the correct animation segments. It will also contain the themable color properties you can set in code. 
+3. Take the output file from LottieGen and place it in your project.
+
+## Design Requirements  
+
+Designers will need to add markers named like the following to all Lottie files that will be used with AnimatedIcon.
+The marker names should reflect the visual state the animation is going from to the visual state the animation is
+going to. For example, the bare minimum of the markers that are needed for icons being used in a Button are:  
+
+* NormalToPointerOver 
+* NormalToPressed 
+* PointerOverToNormal 
+* PointerOverToPressed 
+* PressedToNormal 
+* PressedToPointerOver  
+
+“Normal” is the state where the user is not interacting with the button. 
+
+Markers will need to be named this specific way so AnimatedIcon can map the correct animation segment with the correct visual state transition. This solution is conforming to a standard set of strings that are already defined in the visual state manager for each state. Here is an example of how the markers would look in a Lottie JSON file for a button.  
+
+"markers":[{"tm":0,"cm":"NormalToPointerOver_Start","dr":0},{"tm":9,"cm":"NormalToPointerOver_End","dr":0}, 
+
+{"tm":10,"cm":"NormalToPressed_Start","dr":0},{"tm":19,"cm":"NormalToPressed_End","dr":0}, 
+
+{"tm":20,"cm":"PointerOverToNormal_Start","dr":0},{"tm":29,"cm":"PointerOverToNormal_End","dr":0}, 
+
+{"tm":30,"cm":"PointerOverToPressed_Start","dr":0},{"tm":39,"cm":"PointerOverToPressed_End","dr":0}, 
+
+{"tm":40,"cm":"PressedToNormal_Start","dr":0},{"tm":49,"cm":"PressedToNormal_End","dr":0}, 
+
+{"tm":50,"cm":"PressedToPointerOver_Start","dr":0},{"tm":69,"cm":"PressedToPointerOver_End","dr":0}, 
+
+{"tm":90,"cm":"PressedToNormal_Start","dr":0},{"tm":99,"cm":"PressedToNormal_End","dr":0}, 
+
+{"tm":100,"cm":"PressedToPointerOver_Start","dr":0},{"tm":101,"cm":"PressedToPointerOver_End","dr":0}] 
+
+
+
+# API Pages
+
+## AnimatedIcon class
+
+An AnimatedIcon is a UI component which plays animated images in response to user interaction and visual state changes.
```

Comment:
AnimatedIcon doesn't response to interaction and visual state changes, does it? It has APIs that let you manipulate the animation, and some Xaml controls will call these APIs in response to visual states, and some visual states response to user interaction?

Reply:
Yes that is true and a great clarification. I will update this section to make it clearer.

Reply:
You could say "An AnimatedIcon is a UI component which **enables the playing of** <s>plays</s> animated images in response to user interaction and visual state changes."

Reply:
I still can't figure out how AnimatedIcon is responding to user interaction and visual state changes. Is it getting input events somehow? I thought it had APIs on it that the control would call, but I'm not sure which ones. Does the control code just call SetState to the VSM state?

Reply:
The only thing AnimatedIcon listens to is its state property. So in a precise sense it does not listen to user interaction.  However the scenario we were targetting when designing the control is enabling people to have a lottie icon which responds to user interaction easily.  We accomplish that through VSM setting the state property on AnimatedIcon.

Reply:
We just need more of that precision in the public docs, which is what this is going to become.

Reply:
updated.